modules/nested-elements/assets/js/editor/nested-repeater/views/nested-view-base.js

```diff
@@ -5,7 +5,7 @@ export default class NestedViewBase extends elementor.modules.elements.views.Bas
 		const { elements_placeholder_selector: customSelector, child_container_placeholder_selector: childContainerSelector } = this.model.config.defaults;
 
 		if ( childView !== undefined && childView._index !== undefined && childContainerSelector ) {
-			return containerView.$el.find( `${ childContainerSelector }:nth-child(${ childView._index + 1 })` );
+			return containerView.$el.find( `${ childContainerSelector }` )[ childView._index ];
```

Comment:
The original version wasn't working inside the Mega Menu, because of the multiple level structure.